.github/workflows/main.yml

```diff
@@ -0,0 +1,128 @@
+name: CI
+
+on:
+  push:
+    branches:
+      - master
+      - '*_maintenance'
+  pull_request:
+    branches:
+      - master
+      - '*_maintenance'
+
+jobs:
+
+  build:
+
+    env:
+      # These are required by `config/azure/options.*`, which we are
+      # using temporarily for convenience.
+      # Todo : Ditch Azure and make this workflow as self-contained as
+      # possible.
+      CORTEX_DEPENDENCIES_DIR: build
+      CORTEX_BUILD_NAME: build
+
+    strategy:
+
+      # Don't cancel other jobs in the build matrix if one job fails.
+      fail-fast: false
+
+      matrix:
+
+        # Rather than generate all permutations of various settings,
+        # we want to explicitly list each of the variants we want to
+        # test. We can use `name` to declare the names of our variants,
+        # and then use `include` to define their settings.
+
+        name: [
+          linux-release,
+          linux-debug,
+          macos-release,
+          windows-release
+        ]
+
+        include:
+
+          - name: linux-release
+            os: ubuntu-16.04
+            buildType: RELEASE
+            containerImage: gafferhq/build:1.0.0
+            options: config/azure/options.linux
+
+          - name: linux-debug
+            os: ubuntu-16.04
+            buildType: DEBUG
+            containerImage: gafferhq/build:1.0.0
+            options: config/azure/options.linux
+
+          - name: macos-release
+            os: macos-10.15
+            buildType: RELEASE
+            containerImage:
+            options: config/azure/options.macos
+
+          - name: windows-release
+            os: windows-2019
+            buildType: RELEASE
+            options: config/azure/options.windows
+
+    runs-on: ${{ matrix.os }}
+
+    container: ${{ matrix.containerImage }}
+
+    steps:
+
+    - uses: actions/checkout@v1
+
+    - name: Setup python (Windows)
+      # Revert to Python 2.7 on Windows, for running
+      # things like `installDependencies.py`.
+      uses: actions/setup-python@v1
+      with:
+        python-version: 2.7
+      if: runner.os == 'Windows'
+
+    - uses: ilammy/msvc-dev-cmd@v1
+
+    - name: Install toolchain (Windows)
+      run: python -m pip install scons
+      if: runner.os == 'Windows'
+
+    - name: Install toolchain (MacOS)
+      # Prefer `pip install` because it is faster
+      # than `brew install`.
+      run: sudo pip install scons
+      if: runner.os == 'macOS'
+
+    - name: Install toolchain (Linux)
+      run: |
+        echo "::add-path::/opt/rh/devtoolset-6/root/bin"
+      if: runner.os == 'Linux'
```

Comment:
> What requires this? I hoped the docker container should come ready with devtools 6 set up. Is this something that needs fixing there?

I don't know, that stuff in the Dockerfile is voodoo to me. All I can say for sure is that it doesn't work at all with GitHub actions. This approach works fine for now - I suggest we revisit when Azure is out of the picture and we only have one master to serve.

> Do we no longer need libSegFault.so?

It's a nice-to-have. I'll refine this setup over time, but right now I'm happy with where I'm at (and not with how long it took to get there).

Reply:
Ahh good to know, thanks. 

> I don't know, that stuff in the Dockerfile is voodoo to me.

Yeah, its a bit of a juggle. I think it was necessary because Azure didn't support a container entry point, so it might not be required post-azure. I can add it to my list to look into.